backend/src/feeds/feed.service.ts

```diff
@@ -14,10 +13,15 @@ export class FeedService {
         this.logger = new Logger('FeedService');
     }
 
-    async getFeeds(type: string, limit: number) {
+    async getFeeds(type: string, limit: number, feedId: number) {
         try {
-            const feeds = type ? await this.getFeedsByType(type, limit) : await this.getAll(limit);
-            return feeds.sort((a, b) => compareDesc(a.pub_date, b.pub_date)).slice(0, limit);
+            const feeds = type ?
+                await this.getFeedsByType(type, limit)
+                : (
+                    feedId ?
+                        await this.getMoreByDate(limit, feedId) :
+                        await this.getAll(limit));
+            return feeds.slice(0, limit);
```

Comment:
instead of 0 you should have the skip here.
Also have logic to make sure that skip poiter does not go outOfBound

Reply:
0 is needed here, since the fetching logic is done based on skip.
I have added the logic to make sure that it doesnt go out of bounds